python/tvm/relay/frontend/onnx.py

```diff
@@ -944,6 +946,36 @@ def _impl_v1(cls, inputs, attr, params):
         return Gelu._impl_v1([inp], attr, params)
 
 
+class LayerNormalization(OnnxOpConverter):
```

Comment:
Is there a reason for not using `nn.layer_norm`?

Reply:
The main reason is `nn.layer_norm` doesn't output the intermediate expr such as `mean` and `inv_stdev` needed for onnx output which are about half of the computation. Besides, I first tried with the existing utility `layer_norm` in onnx.py used in `EmbedLayerNormalization` and `SkipLayerNormalization` but that wasn't correct for the `LayerNormalization` because of the wide axes setup so had to reimplement based on onnx spec. I think there should be refactoring opportunity that I'll make a note for later.

Reply:
Do the tests themselves test for these outputs? In the spec they are listed as optional

LayerNorm, much like BatchNorm has different behavior between training and inference time. As ONNX frontend is pretty much for inference, I would say it's ok to drop the other two outputs if it just simplifies the code a lot.

Reply:
Yes the tests, test for these outputs and can't be set as a placeholder